### ISSUE TITLE
Datetime format on departing time

### ISSUE THREAD

**Issue by KHwong12** [Chrisjb/RDistanceMatrix#3]
I do not have much experience on handling datatime data. However, my understanding is that `as.POSIXct()` will convert the time to unix time based on the time zone of the local computer (same as `Sys.timezone()`). Thus the output time in my computer is in  CST (UTC+8), yet not local UK time (UTC+0) and the stop message looks misleading to me.

https://github.com/Chrisjb/RDistanceMatrix/blob/fa208ae3765fcae1f93fe75ed4f8ed31cb4c66dd/R/utils-isochrone.R#L11-L24

Would it be better to require users to input the timezone? We could still, by default, assume the area of analysis is in the same time zone with the computer. (Would it also be better to use the **lubridate** package to handle time?)

```
departing_unix_time <- as.numeric(ymd_hms(departing, tz = Sys.timezone()))

departure_time <- paste0("&departure_time=", departing_unix_time , "&traffic_model=", model))
```

As a side note, instead of using `grepl`, `strptime` may be a better function to validate time input

**Comment by Chrisjb** [comment]
Thanks for this. I would assume that the time entered as the `departing` argument should be local time (for the country you are fetching the travel times for) although I can't find that explicitly in the documentation. 

The warning message is only really appropriate for locations within the UK and setting to the user's timezone wouldn't work for occasions when the user is fetching data outside of their own country. I'll probably remove the error message and think of another way to handle this.

**Comment by Chrisjb** [comment]
Added `tz` argument as suggested which is the best workaround I think. Open another issue if this doesn't fix for you. Thanks!

```r
library(RDistanceMatrix)

us_df <- tibble::tribble(
 ~ origin,           ~destination,
"Central Park, NY", 'One World Trade Center'
)

get_distance(us_df, origin, destination,departing = '2021-01-25 17:00:00', tz= "America/New_York")
#> # A tibble: 1 x 4
#>   origin           destination            driving_distance driving_time
#>   <chr>            <chr>                             <dbl>        <dbl>
#> 1 Central Park, NY One World Trade Center             9291         25.2

get_distance(us_df, origin, destination,departing = '2021-01-25 17:00:00', tz = 'Europe/London')
#> Error in distance_to_destinations(origin = origins, dest = destinations, : departure_time is in the past. Traffic information is only available for future and current times.
```

**Comment by KHwong12** [comment]
Many thanks for the update. Unfortunately, I got this error when using `make_isochrone()`.

```
Error in (function (method, mapbox_api_key, google_api_key, direction, : unused argument (tz = "Europe/London")
```

Looks to me that a `tz` argument needs to be added to the `check_request_valid()` function to bypass the `do.call()`. I remember I used the same method to bypass checking last time (Sorry that I do not know how to handle with `do.call()`).

https://github.com/Chrisjb/RDistanceMatrix/blob/ff30d82a20e125fc82482d29cc1722b246af8d08/R/make_isochrone.R#L61

https://github.com/Chrisjb/RDistanceMatrix/blob/ff30d82a20e125fc82482d29cc1722b246af8d08/R/utils-isochrone.R#L184
